### DISCUSSION TITLE
Change channel before Checkout (and update prices)

### DISCUSSION THREAD

**Discussion by slamer59** [saleor/saleor#14206] [Q&A] [answered]
Hello,
I have two channels FR and SN. First in € and Second in XOF (FCFA). 
I have two gateway Stripe and Wave Money (West African).

In the UI I want to present the two payments whatever the channel. 
My problem : 
- the user add to cart in SN channel in FCFA and pays with Stripe (in euro cent). 

My first guess is to convert the cart from SN to FR which is in Euro. 
How to convert it ? 
I want to convert the cart from FCFA to Euro. 

Is there a graphql call which can update the channel directly ? Or should I construct some logic to remove SN and add the product in the other channel ? 

Regards

**Comment by peteror** [comment]
I had the same issue last week - there is no way to change the checkout's channel through the API. I ended up writing code that took products from a checkout and added them to a new checkout on the other channel. (plus some error handling in case some products are not available / published on the new channel)

**Reply by timuric** [reply, marked as answer]
Converting basket from one channel to another is not possible via API because there could be many edge cases such as different stock and availability of the products well as regional constraints. Overall channels are designed to serve different markets rather than compete with each other.

Another aspect to consider is price arbitrage, having static prices in catalog would render specific currency more attractive given the exchange rate of the day, therefore you would want to use dynamic currency conversion.


One way to allow customers to pay in different currencies from a single channel is to use payment gateway multi currency capabilities which would give dynamic rate and guarantee that you as a business would receive desired value in base currency.

**Reply by slamer59** [reply]
Well, arguments are good. 
But for the stock it is for my point of view related to warehouse and not market.

I end up into creating a new checkout on the fly, recreate lines to cart. And thats it ! I think I will write this in checkout app as an API endpoint to make majority of API call on the server side.

Thanks ! At least I will not search for something which doesn't exist for my special case.
